subsys/bluetooth/host/l2cap.c

```diff
@@ -1692,6 +1692,19 @@ static void l2cap_chan_le_recv(struct bt_l2cap_le_chan *chan,
 
 	l2cap_chan_send_credits(chan, buf, 1);
 }
+
+static void l2cap_chan_recv_queue(struct bt_l2cap_le_chan *chan,
+				  struct net_buf *buf)
+{
+	if (chan->chan.state == BT_L2CAP_DISCONNECT) {
+		BT_WARN("Ignoring data received while disconnecting");
+		net_buf_unref(buf);
+		return;
+	}
```

Comment:
Don't we want this check somewhere earlier and applied also the fixed channels?

Reply:
Fixed channels are never in disconnect, they are either connected or disconnected.